src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java

```diff
@@ -145,79 +141,158 @@ private static int getUserCount(Request req, Response res) {
-                "\"connection\": \"Username-Password-Authentication\"," +
-                "\"email\": %s," +
-                "\"password\": %s," +
-                "\"app_metadata\": {\"datatools\": [%s] } }"
-                , jsonNode.get("email"), jsonNode.get("password"), jsonNode.get("permissions"));
-        setRequestEntityUsingJson(createUserRequest, json, req);
+    /**
+     * Class to hold a user's profile and raw JSON permissions.
+     */
+    public static class Auth0UserSearchResult {
+        /**
+         * If the tenant user already has permissions related to another instance of datatools they are preserved here.
+         * This will prevent them from being overwritten when creating permissions for other datatools instances.
+         */
+        String existingPermissions = "";
 
-        return executeRequestAndGetResult(createUserRequest, req);
+        /**
+         * Used to hold email, userId and expanded datatools parameters if they match the AUTH0_CLIENT_ID parameter.
+         */
+        Auth0UserProfile user;
     }
 
+    /**
+     * Update user permissions. Only used in cases where the Auth0 user already exists and permissions for the current
+     * instance of datatools has not been set previously.
+     */
+//    private static String updateUserPermissions(Request req, Auth0UserSearchResult searchResult, String newPermissions) {
+//        // if there are existing permissions the assumption is that the datatools info has been set, but, only if it
+//        // matches the current instance of datatools (AUTH0_CLIENT_ID).
+//        if(!searchResult.existingPermissions.isEmpty() && searchResult.user.getApp_metadata().getDatatoolsInfo() != null) {
+//            LOG.info("Permissions for this Auth0 user {} already exist for this instance {} of datatools",
+//                searchResult.user.getEmail(), AUTH0_CLIENT_ID);
+//            logMessageAndHalt(
+//                req,
+//                400,
+//                String.format("User %s already exists", searchResult.user.getEmail())
+//            );
+//        }
+//
+//        LOG.info("Auth0 user {} already exists, updating permissions", searchResult.user.getEmail());
+//        // User is unknown to this instance (AUTH0_CLIENT_ID) of datatools, update permissions, making sure to
+//        // preserve any previously held permissions.
+//        String permissions = (searchResult.existingPermissions.isEmpty()) ?
+//            newPermissions : String.format("%s,%s", searchResult.existingPermissions, newPermissions);
+//        String json = String.format("{ \"app_metadata\": { \"datatools\" : [%s]  }}",permissions);
+//        HttpPatch updateUserRequest = new HttpPatch(getUserIdUrl(req, searchResult.user.getUser_id()));
+//        setHeaders(req, updateUserRequest);
+//        setRequestEntityUsingJson(updateUserRequest, json, req);
+//        return executeRequestAndGetResult(updateUserRequest, req);
+//    }
+
```

Comment:
I'm not sure why the above is just commented out and not deleted.

Reply:
Updated.